ietf/community/models.py

```diff
@@ -30,8 +39,8 @@ def __str__(self):
 
     def get_absolute_url(self):
         import ietf.community.views
-        if self.user:
-            return urlreverse(ietf.community.views.view_list, kwargs={ 'username': self.user.username })
+        if self.person:
+            return urlreverse(ietf.community.views.view_list, kwargs={ 'username': self.person.user.username })
```

Comment:
removing the exposure of username here is also worth pursuing. We could either change this to just hit the person pk (which I would prefer to avoid) or do the same kind of thing we do at the /person/ endpoint, where we match names or email addresses. Please take a look at doing that and see if there's a strong reason not to.

Reply:
Okay, I'll look into it.

Reply:
This ticket is about refactoring, which should be (mostly) invisible to the user. Changing the URL structure is not only user-visible, but goes way beyond the scope of this ticket. Which is not to say it's not worth doing, but it should be its own ticket.